my-sms/spec/models/student_spec.rb

```diff
@@ -1,5 +1,66 @@
+# frozen_string_literal: true
+
 require 'rails_helper'
 
-RSpec.describe Student, :type => :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+RSpec.describe Student, type: :model do
+  it 'has a valid factory' do
+    expect(FactoryBot.build(:student)).to be_valid
+  end
+
+  it 'has a factory that generates unique emails' do
+    student = FactoryBot.build(:student)
+    other_student = FactoryBot.build(:student)
+    expect(student.email).not_to eq other_student.email
+  end
+
+  it 'is invalid with a duplicate email address' do
+    FactoryBot.create(:student, email: 'user@example.com')
+    student = FactoryBot.build(:student, email: 'user@example.com')
+    student.valid?
+    expect(student.errors[:email]).to include('has already been taken')
+  end
+
+  it 'is invalid without a first name' do
+    student = FactoryBot.build(:student, first_name: nil)
+    student.valid?
+    expect(student.errors[:first_name]).to include('can\'t be blank')
```

Comment:
We generally prefer to use double quotes over single if that means you can avoid escaping apostrophes like this. It's good form to use single quotes whenever you're not using string interpolation, so I like that you've done that, but I would generally just use `"can't be blank"` in this case.

Reply:
Understood. This ended up removed anyway in f45e16ad5f92b912d1aeee870101c3a8026d406f